File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/Webhook.kt
package com.example.kotlinmpplib

import breez_sdk.*
class Webhooks {
    fun registerWebhook(sdk: BlockingBreezServices) {
        // ANCHOR: register-webook
        try {
            val url = "https://your-nds-service.com/api/v1/notify?platform=<ios|android>&token=<PUSH_TOKEN>"
            sdk.registerWebhook(url)
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: register-webook
    }

    fun unregisterWebhook(sdk: BlockingBreezServices) {
        // ANCHOR: unregister-webook
        try {
            val url = "https://your-nds-service.com/api/v1/notify?platform=<ios|android>&token=<PUSH_TOKEN>"
            sdk.unregisterWebhook(url)
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: unregister-webook
    }
}

File: snippets/swift/BreezSDKExamples/Sources/Webhook.swift
//
//  ServiceStatus.swift
//  
//
//

import Foundation
import BreezSDK

func registerWebhook(sdk: BlockingBreezServices) throws {
    // ANCHOR: register-webook
    let url = "https://your-nds-service.com/notify?platform=<ios|android>&token=<PUSH_TOKEN>"
    try sdk.registerWebhook(webhookUrl: url)  
    // ANCHOR_END: register-webook
}

func unregisterWebhook(sdk: BlockingBreezServices) throws {
    // ANCHOR: unregister-webook
    let url = "https://your-nds-service.com/notify?platform=<ios|android>&token=<PUSH_TOKEN>"
    try sdk.unregisterWebhook(webhookUrl: url)  
    // ANCHOR_END: unregister-webook
}
